asyncscope.md

```diff
@@ -1094,6 +1162,31 @@ requirement to complete on the receiver's scheduler restricts which receivers a
 exchange for determinism; the alternative would have the _`join-sender`_ completing on the execution context of
 whichever nested operation happens to be the last one to complete.
 
+## When to use `counting_scope` vs `let_with_async_scope`
+
+Although `counting_scope` and `let_with_async_scope` have overlapping use-cases, we specifically designed the two
+facilities to address separate problems. In short, `counting_scope` is best used in an unstructured context and `let_with_async_scope`
+is best used in a structured context.
+
+We define "unstructured context" as:
+- a place where using `sync_wait` would be inappropriate, 
+- and you can't "solve by induction" (i.e you're not in an async context where you can start the sender by "awaiting" it)
+
+`counting_scope` should be used when you have a sender you want to start in an unstructured context. In this case, `spawn(sender, scope)`
+ would be the preferred way of starting asynchronous work. `scope.join()` needs to be called upon the owning object's destruction
+in order to ensure that the object's lifetime lives at least until all asynchronous work completes. Note that exception safety
+needs to be handled explicitly in the use of `counting_scope`.
+
+`let_with_async_scope` returns a sender, and therefore can only be started in one of 3 ways:
+1. `sync_wait`
+2. `spawn` on a `counting_scope`
+3. await
+
+`let_with_async_scope` will manage the scope for you which, as a result, won't give you the ability to manually manage the lifetime of
+resources the way `counting_scope` does. This behavior is a feature, since the scope being managed by `let_with_async_scope` is intended
+to live only until the sender completes. This also means that `let_with_async_scope` will be exception safe by default.
```

Comment:
> won't give you the ability to manually manage the lifetime of resources the way `counting_scope` does

I'm worried this comes off as a negative thing. Your next sentence is an explanation of why it's a feature, but I'm wondering if there's a wording that makes it sound positive from the start. I'm not sure what to suggest, but here's an attempt:
```suggestion
`let_with_async_scope` will manage the scope for you, ensuring that the managed scope is always joined before
`let_with_async_scope` completes.  The algorithm frees the user from having to manage the coupling between the
lifetimes of the managed scope and the resource(s) it protects at the cost of having to describe the nested work in
a fully structured way.
```
or
```suggestion
`let_with_async_scope` will manage the scope for you, ensuring that the managed scope is always joined before
`let_with_async_scope` completes.  The algorithm frees the user from having to manage the coupling between the
lifetimes of the managed scope and the resource(s) it protects with the limitation that the nested work must be
fully structured.
```